unwrap bl children when passed to append() & support bl in constructor

rvagg opened the pull request:
fixes: https://github.com/rvagg/bl/issues/26

Another improvement on this would be to move the array checking stuff down in to `append()` so even it will take arrays of stuff like the constructor.

/R=@mcollina

rvagg commented on bl.js:
```diff
@@ -49,16 +47,21 @@ BufferList.prototype._offset = function (offset) {
     buf = buf.toString()
 
-  this._bufs.push(isBuffer ? buf : new Buffer(buf))
-  this.length += buf.length
```
pretty sure there was a bug in here if you pass in a string with some unicode multibyte, in fact we don't really do any unicode testing so who knows what's lurking in here

mcollina replied:
I agre!

mcollina commented on bl.js:
```diff
@@ -49,16 +47,21 @@ BufferList.prototype._offset = function (offset) {
+    // unwrap argument into individual BufferLists
+    for (var i = 0; i < buf._bufs.length; i++)
+      this.append(buf._bufs[i])
+    return
```
this should be `return this`

mcollina commented:
I also did https://github.com/rvagg/bl/pull/27 :).

This is ok anyway!

rvagg commented:
pushed an update, fixed that `return` bug, introduced more tests for that, made it so the constructor and `append()` accept the same types of arguments (sans callback) so everything is nice and symetrical and everything unrolls nicely.

rvagg commented:
added some notes to the readme, rebased on 1.0.3

I propose we do a 1.1.0 with this

jcrugzz commented:
This looks good :+1:. `1.1.0` sounds reasonable as it introduces more capability to the `append` api.

mcollina commented:
:+1: for it being 1.1.0.

rvagg commented:
done!

rvagg commented:
OK, I actually messed up that merge .. so I've had to publish 1.1.1, I've unpublished 1.1.0 cause it's really just 1.0.3.
